tools/engine_tool/lib/src/commands/lint_command.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' show File;
+
+import 'package:path/path.dart' as p;
+
+import '../dart_utils.dart';
+import '../proc_utils.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'lint' command.
+final class LintCommand extends CommandBase {
+  /// Constructs the 'lint' command.
+  LintCommand({
+    required super.environment,
+  }) {
+    _linterArgs['dart'] = <String>[findDartBinDirectory(environment)];
+    _linterRunner['java'] = p.join(findDartBinDirectory(environment), 'dart');
+  }
+
+  // Map from lint-type to path of tool to invoke.
+  // Paths are relative to 'engine/src/flutter'.
+  static const Map<String, String> _linters = <String, String>{
+    'dart': 'ci/analyze.sh',
+    'python': 'ci/pylint.sh',
+    'c': 'ci/clang_tidy.sh',
+    'java': 'tools/android_lint/bin/main.dart',
+  };
+
+  final Map<String, List<String>> _linterArgs = <String, List<String>>{};
+
+  final Map<String, String> _linterRunner = <String, String>{};
+
+  @override
+  String get name => 'lint';
+
+  @override
+  String get description => 'Lint the engine repository.';
+
+  @override
+  Future<int> run() async {
+    // TODO(loic-sharma): Relax this restriction.
+    if (environment.platform.isWindows) {
+      environment.logger
+          .error('lint command is not supported on Windows (for now).');
```

Comment:
We should consider just adding a `logger.fatal` that:

- Throws `FatalError(...)`
- The root program catches it, displays it (in verbose mode, with a stack trace), and returns 1